squidb-addons/squidb-recyclerview/src/com/yahoo/squidb/recyclerview/SquidRecyclerAdapter.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the Apache 2.0 License.
+ * See the accompanying LICENSE file for terms.
+ */
+package com.yahoo.squidb.recyclerview;
+
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.RecyclerView.Adapter;
+
+import com.yahoo.squidb.data.AbstractModel;
+import com.yahoo.squidb.data.SquidCursor;
+import com.yahoo.squidb.sql.Property;
+
+/**
+ * RecyclerView.Adapter implementation backed by a {@link SquidCursor}. Subclasses must also define an implentation
+ * of {@link SquidViewHolder}, which recycles a SquiDB model instance along with the associated itemView. This
+ * adapter automatically populates the model instance with data from the backing SquidCursor, so that it can be used
+ * to help bind data to the itemView inside of {@link #onBindSquidViewHolder(SquidViewHolder, int)
+ * onBindSquidViewHolder}.
+ *
+ * @param <M> the model type of the backing SquidCursor
+ * @param <V> a SquidViewHolder implementation
+ */
+public abstract class SquidRecyclerAdapter<M extends AbstractModel, V extends SquidViewHolder<? extends M>>
+        extends RecyclerView.Adapter<V> {
+
+    private SquidCursor<M> cursor;
+    private Property<Long> idProperty;
+
+    /**
+     * Construct a new SquidRecyclerAdapter
+     */
+    public SquidRecyclerAdapter() {
+        this(null, null);
+    }
+
+    /**
+     * Construct a new SquidRecyclerAdapter backed by the specified SquidCursor.
+     *
+     * @param cursor the backing SquidCursor
+     */
+    public SquidRecyclerAdapter(SquidCursor<M> cursor) {
+        this(cursor, null);
+    }
+
+    /**
+     * Construct a new SquidRecyclerAdapter that uses the specified column to determine item IDs in {@link
+     * #getItemId(int)}. This should be a column that is distinct and non-null for every row in the cursor. If this
+     * argument is not null, the adapter will report that it has stable item IDs (see {@link Adapter#hasStableIds()
+     * hasStableIds()}).
+     *
+     * @param idProperty the column to use for item IDs
+     */
+    public SquidRecyclerAdapter(Property<Long> idProperty) {
+        this(null, idProperty);
+    }
+
+    /**
+     * Construct a new SquidRecyclerAdapter backed by the specified SquidCursor and that uses the specified column to
+     * determine item IDs in {@link #getItemId(int)}. This should be a column that is distinct and non-null for every
+     * row in the cursor. If this argument is not null, the adapter will report that it has stable item IDs (see {@link
+     * Adapter#hasStableIds() hasStableIds()}).
+     *
+     * @param cursor the backing SquidCursor
+     * @param idProperty the column to use for item IDs
+     */
+    public SquidRecyclerAdapter(SquidCursor<M> cursor, Property<Long> idProperty) {
+        this.cursor = cursor;
+        this.idProperty = idProperty;
+        if (idProperty != null) {
+            setHasStableIds(true);
```

Comment:
In SquidCursorAdapter, we say hasStableIds is always true and fall back to using _id to get a row id if the user doesn't specify a different property to use. Should we be consistent across both adapters how we handle this?